pkg/manager/member/tidb_discovery_manager.go

```diff
@@ -47,6 +47,10 @@ func NewTidbDiscoveryManager(typedControl controller.TypedControlInterface) Tidb
 
 func (m *realTidbDiscoveryManager) Reconcile(tc *v1alpha1.TidbCluster) error {
 
+	// If discovery is not specified return
+	if tc.Spec.Discovery == nil {
```

Comment:
Currently, the `Spec.Discovery` only specifies the resource requirement of discovery component, actually, the creation of discovery is not controlled by the `Spec.Discovery`, I think we can check the `spec.pd ==nil` here for this PR.